test/support/utils_sort.h

```diff
@@ -0,0 +1,473 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _UTILS_SORT_H
+#define _UTILS_SORT_H
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#include <random>
+#include <atomic>
+#include <string>
+#include <vector>
+#include <utility>
+#include <cstdint>
+#include <type_traits>
+
+// TODO: check if still needed
+#define _CRT_SECURE_NO_WARNINGS
+
+#include "utils.h"
+#if TEST_DPCPP_BACKEND_PRESENT
+#    include "support/sycl_alloc_utils.h"
+#endif
+
+// The struct allows filtering configurations to avoid exponential growth of test cases
+// and to pass an error message prefix for better diagnostics
+struct SortTestConfig
+{
+    bool is_stable;
+    // Ignored for host policies
+    bool test_usm_device;
+    bool test_usm_shared;
+
+    std::string err_msg_prefix;
+
+    SortTestConfig(bool is_stable, bool usm_device = false, bool usm_shared = false, const std::string& msg = "")
+        : is_stable(is_stable), test_usm_device(usm_device), test_usm_shared(usm_shared), err_msg_prefix(msg) {}
+
+    SortTestConfig msg(const std::string& msg) const
```

Comment:
Looks like UB: you pass 3 uninitialized fields into constructor: `is_stable, test_usm_device, test_usm_shared`